ethstats/ethstats.go

```diff
@@ -259,81 +306,104 @@ func (s *Service) loop(ctx context.Context) {
 				urls = []string{"wss://" + path, "ws://" + path}
 			}
 
-			// Establish a websocket connection to the server on any supported URL
-			var (
-				conn *websocket.Conn
-				err  error
-			)
-
+			errTimer := time.NewTimer(0)
+			defer errTimer.Stop()
 			// Loop reporting until termination
 			for {
-				dialer := websocket.Dialer{HandshakeTimeout: 5 * time.Second}
-				header := make(http.Header)
-				for _, url := range urls {
-					conn, _, err = dialer.Dial(url, header)
-					if err == nil {
-						break
+				select {
+				case <-ctxGroup.Done():
+					return ctxGroup.Err()
+				case <-errTimer.C:
+					var (
+						conn *connWrapper
+						err  error
+					)
+					// Establish a websocket connection to the server on any supported URL
+					dialer := websocket.Dialer{HandshakeTimeout: 5 * time.Second}
+					header := make(http.Header)
+					header.Set("origin", "http://localhost")
+					for _, url := range urls {
```

Comment:
This isn't new, but it is it correct that we only use one connection?

Reply:
To the celostats server? I think that as is not critical data, handling only one connection keeps the other code (and even the celostats server) simpler.

Reply:
Yea. I agree it makes it simpler, but it's weird to have multiple urls but only use 1